disambiguator.py

```diff
@@ -371,95 +385,143 @@ def group_categories():
+    tokens = data[:, 0]
+    labels = data[:, 1]
+
+    # get features
+    darrays = get_descriptor_arrays(tokens)
+    # labels = get_dummies(labels)
+
+    # put into grams (give context)
+    darrays, labels = make_grams(
+        darrays, labels, 3, target_tag=EOS_PUNC)
+
+    (tr_inputs, te_inputs), (tr_outputs, te_outputs) = ctd.split_data(
+        darrays, out_data=labels, frac=0.80)
+
+    if save_to_disk:
+        np.save('storage/data_nn_disambiguator/X_train.npy', tr_inputs)
+        np.save('storage/data_nn_disambiguator/X_test.npy', te_inputs)
+        np.save('storage/data_nn_disambiguator/y_train.npy', tr_outputs)
+        np.save('storage/data_nn_disambiguator/y_test.npy', te_outputs)
+
+    return (tr_inputs, te_inputs), (tr_outputs, te_outputs)
+
+
+def main():
+    ''' Paper found parameters to be efficient:
+        6 word contexts --> 3 num_grams
+        2 hidden_units
+    '''
+
+    X_train = np.load('storage/data_nn_disambiguator/X_train.npy')
+    X_test = np.load('storage/data_nn_disambiguator/X_test.npy')
+    y_train = np.load('storage/data_nn_disambiguator/y_train.npy')
+    y_test = np.load('storage/data_nn_disambiguator/y_test.npy')
+
+    trained_weights = nn.train_nn(
+        X_train, y_train, [50, 10],
         batch_size=256, param_scale=0.1,
-        num_epochs=5, step_size=0.001, L2_reg=1.0):
-
-    num_input_dims = tr_obs_set.shape[1]
-    layer_sizes = [num_input_dims, num_hiddens, 1]
-    init_params = init_random_params(param_scale, layer_sizes)
-    num_batches = int(np.ceil(tr_obs_set.shape[0] / batch_size))
-
-    def batch_indices(iter):
-        idx = iter % num_batches
-        return slice(idx * batch_size, (idx+1) * batch_size)
-
-    # Define training objective
-    def objective(params, iter):
-        idx = batch_indices(iter)
-        return -log_posterior(
-            params, tr_obs_set[idx], tr_out_set[idx], L2_reg)
-
-    # Get gradient of objective using autograd.
-    objective_grad = grad(objective)
-
-    print("     Epoch     |    Train accuracy  |       Test accuracy  ")
-
-    def print_perf(params, iter, gradient):
-        if iter % num_batches == 0:
-            train_acc = accuracy(params, tr_obs_set, tr_out_set)
-            test_acc = accuracy(params, test_images, test_labels)
-            print("{:15}|{:20}|{:20}".format(
-                iter//num_batches, train_acc, test_acc))
-
-    # The optimizers provided can optimize lists, tuples, or dicts of
-    # parameters.
-    optimized_params = adam(
-        objective_grad, init_params, step_size=step_size,
-        num_iters=num_epochs * num_batches, callback=print_perf)
-
-    return optimized_params
+        num_epochs=20, step_size=0.001, L2_reg=1.0)
+
+    # save the weights
+    np.save('storage/trained_weights.npy', trained_weights)
+
+    y_pred = nn.neural_net_predict(trained_weights, X_test)
+    # don't forget to exp
+    print("auc: {}".format(
+        ctd.get_auc(y_test[:, 1], np.exp(y_pred))))
```

Comment:
Why's it `y_test[:, 1]`, shouldn't it be 0? Isn't it just a 1-D array? This depends on what you were hoping to go for in the `make_grams`, so could be a no-op